Mlem/Extensions/JSONDecoder+Default.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  JSONDecoder+Default.swift
+//  Mlem
+//
+//  Created by Nicholas Lawson on 11/06/2023.
+//
+
+import Foundation
+
+extension JSONDecoder {
+    static var defaultDecoder: JSONDecoder {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        
+        let formatter = DateFormatter()
+        
+        decoder.dateDecodingStrategy = .custom({ decoder in
+            let container = try decoder.singleValueContainer()
+            let string = try container.decode(String.self)
+            
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS"
+            if let date = formatter.date(from: string) {
+                return date
+            }
+            
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+            if let date = formatter.date(from: string) {
+                return date
+            }
```

Comment:
You're missing the not-T variant as described [here](https://github.com/buresdv/Mlem/pull/95#discussion_r1225766951), otherwise its good

Reply:
I think this (b6ca0ec6a8d80bf3a3f47a200393c75d8d6f744c) is what you mean? I may be mistaken? 😅

Reply:
Yes, this is good